script.skipit/resources/lib/skipit.py

```diff
@@ -0,0 +1,172 @@
+import xbmc
+import xbmcgui
+import xbmcaddon
+import threading
+
+# create a class for your addon, we need this to get info about your addon
+ADDON = xbmcaddon.Addon()
+# get the full path to your addon, decode it to unicode to handle special (non-ascii) characters in the path
+CWD = ADDON.getAddonInfo('path')
+ 
+skip_secs = 180 
+min_skip_secs = 3 
+slippage = 1
+timeout = 5.0
+going_forward = False
+going_backward = False
+
+class SkipItWindow(xbmcgui.WindowXMLDialog):
+
+    # until now we have a blank window, the onInit function will parse your xml file
+    def onInit(self):
+        
+        global my_timer
+        global skip_secs
+        global min_skip_secs
+        global slippage
+        global timeout
+
+        # select a view mode, '50' in our case, as defined in the skin file
+        xbmc.executebuiltin('Container.SetViewMode(50)')
```

Comment:
Which skin file? I don't think this is allowed per our rules. Add-ons should not force view types on skins

Reply:
I'll delete this. I just copied from the "Hello World" tutorial.